src/coreclr/jit/valuenum.cpp

```diff
@@ -8753,9 +8753,9 @@ void Compiler::fgValueNumberIntrinsic(GenTree* tree)
     }
     else
     {
-        switch (intrinsic->gtIntrinsicId)
+        switch (intrinsic->gtIntrinsicName)
```

Comment:
Seems weird to have `switch` with just one `case` and default asserting with `unreached()`. 
Should this be replaced with `assert(intrinsic->gtIntrinsicName == NI_System_Object_GetType)` ?